js/interface.js

```diff
@@ -261,35 +280,180 @@ function getFolderContentsById(id, type, isSearchNav) {
+      })
+    } else {
+      restorePromise = Fliplet.API.request({
+        url: 'v1/media/files/' + itemID + '/restore',
+        method: 'POST',
+      }).then(function() {
+        $element.remove();
+        checkboxStatus();
+        completedItems++;
+
+        currentFiles = currentFiles.filter(function(file){
+          return file.id != itemID;
+        });
+
+        $('.file-table-header input[type="checkbox"]').prop('checked', false);
+      })
+    }
+    restorePromise.then(function(result) {
+      showSpinner(false);
+      if (completedItems === items.length && completedItems !== 1) {
+        Fliplet.Modal.alert({
+          title: 'Restore complete',
+          message: items.length + ' items restored'
+        })
+      } else if (completedItems === items.length && completedItems === 1) {
+        Fliplet.Modal.confirm({
+          title: 'Restore complete',
+          message: itemName + ' restored',
+          buttons: {
+            cancel: {
+              label: 'Go to folder',
+              className: 'btn-default nav-folder'
+            },
+            confirm: {
+              label: 'OK',
+              className: 'btn-primary'
+            },
+          },
+        }).then(function(result) {
+          if (!result) {
+            navigateToFolder($element);
+          }
+        })
+      }
+    }).catch(function(error) {
+      showSpinner(false);
+      Fliplet.Modal.alert({
+        title: 'Restore failed',
+        message: Fliplet.parseError(error),
+      })
+    })
+  })
+}
+
+function removeTrashItems(items) {
+  completedItems = 0;
+
+  $(items).each(function() {
+    var $element = $(this);
+    var itemID = $element.attr('data-id');
+    var itemName = $element.attr('data-name');
+    var deletePromise;
+
+    showSpinner(true);
+
+    if ($element.attr('data-file-type') === 'folder') {
+      deletePromise = Fliplet.API.request({
+        url: 'v1/media/deleted/folders/' + itemID,
+        method: 'DELETE',
+      }).then(function() {
+        $element.remove();
+        checkboxStatus();
+        completedItems++;
+
+        currentFolders = currentFolders.filter(function(folder){
+          return folder.id != itemID;
+        });
+        $('.file-table-header input[type="checkbox"]').prop('checked', false);
+      })
+    } else {
+      deletePromise = Fliplet.API.request({
+        url: 'v1/media/deleted/files/' + itemID,
+        method: 'DELETE',
+      }).then(function() {
+        $element.remove();
+        checkboxStatus();
+        completedItems++;
+
+        currentFiles = currentFiles.filter(function(file){
+          return file.id != itemID;
+        });
+
+        $('.file-table-header input[type="checkbox"]').prop('checked', false);
+      })
+    }
+    deletePromise.then(function(result) {
+      showSpinner(false);
+      if (completedItems === items.length && completedItems !== 1) {
+        Fliplet.Modal.alert({
```

Comment:
This logic seems repeated with the only difference of the actual message in the alert. Please refactor to use a variable for the message.